File: Offer/2269/main.go
package main

type ListNode struct {
	Val  int
	Next *ListNode
}

func getIntersectionNode(headA, headB *ListNode) *ListNode {
	hashmap := make(map[*ListNode]bool)
	for headA != nil {
		hashmap[headA] = true
		headA = headA.Next
	}
	var ok bool
	for headB != nil {
		if _, ok = hashmap[headB]; ok {
			return headB
		}
		headB = headB.Next
	}
	return nil
}

File: Offer/2269/main.py
from typing import Dict


class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> ListNode:
        hashmap: Dict[ListNode, bool] = {}
        while headA != None:
            hashmap[headA] = True
            headA = headA.next
        while headB != None:
            if hashmap.get(headB) != None:
                return headB
            headB = headB.next
        return None